Mathlib/RingTheory/Binomial.lean

```diff
@@ -72,13 +72,31 @@ theorem factorial_nsmul_multichoose_eq_eval_ascPochhammer (r : R) (n : ℕ) :
     n.factorial • multichoose r n = Polynomial.eval r (ascPochhammer R n) :=
   BinomialRing.factorial_nsmul_multichoose r n
 
+end Ring
+
 instance Nat.instBinomialRing : BinomialRing ℕ where
   nsmul_right_injective n hn r s hrs := Nat.eq_of_mul_eq_mul_left (Nat.pos_of_ne_zero hn) hrs
   multichoose := Nat.multichoose
   factorial_nsmul_multichoose r n := by
     rw [Nat.multichoose_eq, smul_eq_mul, ← Nat.descFactorial_eq_factorial_mul_choose,
     ascPochhammer_nat_eq_descFactorial]
 
-end Ring
+instance Int.instBinomialRing : BinomialRing ℤ where
+  nsmul_right_injective n hn r s hrs := Int.eq_of_mul_eq_mul_left (Int.ofNat_ne_zero.mpr hn) hrs
+  multichoose r k := by
+    cases r with
+    | ofNat n =>
+      use ((Nat.choose (n + k - 1) k):ℤ)
+    | negSucc n => use (-1)^k * Nat.choose n.succ k
```

Comment:
In practice, it turns out that it is better to factor out such definitions-by-cases into standalone functions. They are complicated to work with, when defined as a field of an instance because of their non-trivial definitional equalities.

So I suggest creating a definition `Int.multichoose` above this instance, with the definition of this function. And then this field just refers to that definition.

Reply:
I see.  Thank you for that suggestion.